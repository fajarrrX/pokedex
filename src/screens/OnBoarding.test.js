import React from "react";
import renderer from "react-test-renderer";
import OnBoardingScreen from "./OnBoardingScreen";

it("OnBoardingScreen screen renders correctly", () => {
  renderer.create(<OnBoardingScreen />);
});

test("OnBoardingScreen snapShot", () => {
  const snap = renderer.create(<OnBoardingScreen />).toJSON();
  expect(snap).toMatchSnapshot();
});
