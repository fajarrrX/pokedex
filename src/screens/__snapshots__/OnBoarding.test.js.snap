// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnBoardingScreen snapShot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#a6e4d0",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "backgroundColor": "#a6e4d0",
          "image": <Image
            source={1}
          />,
          "subtitle": "A New Way To Connect With The Pokemon",
          "title": "Connect to the Pokedex",
        },
        Object {
          "backgroundColor": "#e9bcbe",
          "image": <Image
            source={1}
          />,
          "subtitle": "Pick Your Favorites Pokemon to Your Team",
          "title": "Show Your Favorites",
        },
        Object {
          "backgroundColor": "#fdeb93",
          "image": <Image
            source={1}
          />,
          "subtitle": "Let The Spot Light Capture You",
          "title": "Become The Star",
        },
      ]
    }
    disableVirtualization={false}
    extraData={null}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={1}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfig={
      Object {
        "itemVisiblePercentThreshold": 100,
      }
    }
    viewabilityConfigCallbackPairs={
      Array [
        Object {
          "onViewableItemsChanged": [Function],
          "viewabilityConfig": Object {
            "itemVisiblePercentThreshold": 100,
          },
        },
      ]
    }
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "paddingTop": 0,
              },
              null,
              Object {
                "height": 1334,
                "width": 750,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 0,
                  "paddingBottom": 60,
                  "width": "100%",
                },
                null,
              ]
            }
          >
            <Image
              source={1}
            />
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              allowFontScaling={true}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 26,
                    "paddingBottom": 15,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#000",
                  },
                  Object {},
                ]
              }
            >
              Connect to the Pokedex
            </Text>
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
              }
            }
          >
            <Text
              allowFontScaling={true}
              style={
                Array [
                  Object {
                    "color": "rgba(255, 255, 255, 0.7)",
                    "fontSize": 16,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.7)",
                  },
                  Object {},
                ]
              }
            >
              A New Way To Connect With The Pokemon
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "width": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.1)",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-between",
          "paddingHorizontal": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexShrink": 1,
            "width": 200,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginHorizontal": 10,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Skip
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.8)",
              "height": 6,
              "marginHorizontal": 3,
              "width": 6,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.3)",
              "height": 6,
              "marginHorizontal": 3,
              "width": 6,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.3)",
              "height": 6,
              "marginHorizontal": 3,
              "width": 6,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexShrink": 1,
            "width": 200,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginHorizontal": 10,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Next
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
